File: src/main/java/com/redbox/domain/attach/dto/AttachFileResponse.java
package com.redbox.domain.attach.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.redbox.domain.attach.entity.AttachFile;
import lombok.Getter;

@Getter
@JsonDeserialize
public class AttachFileResponse {
    private Long fileNo;
    private String originFilename;
    private String filename;

    @JsonCreator
    public AttachFileResponse() {
    }

    public AttachFileResponse(AttachFile file) {
        this.fileNo = file.getId();
        this.originFilename = file.getOriginalFilename();
        this.filename = file.getNewFilename();
    }
}

File: migration/src/main/kotlin/com/redbox/domain/community/attach/dto/AttachFileResponse.kt
package com.redbox.domain.community.attach.dto

import com.fasterxml.jackson.annotation.JsonTypeInfo
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.redbox.domain.community.attach.entity.AttachFile

@JsonDeserialize
@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
data class AttachFileResponse(
    var fileNo: Long? = null,
    var originFilename: String? = null,
    var filename: String? = null
) {
    constructor(file: AttachFile) : this(
        fileNo = file.id,
        originFilename = file.originalFilename,
        filename = file.newFilename
    )
}
